### PR TITLE
170766097 Endpoint to reject trip request

### PR DESCRIPTION
**Pull request by Nigorjeanluc**
#### What does this PR do?
This pull request ensure the user's manager can reject a trip request
### Description of Task to be completed?
 - use `server/middlewares/isManager.js` in route to verify if the user's role is a manager
 - add a method into `server/middlewares/statusValidation.js` to validate if the trip's status is pending, approved or rejected.
 - add test into `server/tests/request.test.js` file to this endpoint
#### How should this be manually tested?
You can pull the `develop` or `ft-reject-request-170766097` branch and run these following commands `npm install` and run `npm test` 
#### What are the relevant pivotal tracker stories?
[#170766097](https://www.pivotaltracker.com/n/projects/2429839/stories/170766097)

### CODE REVIEW

**Review by manzi-guev — APPROVED** [review]

**Review by karengir — APPROVED** [review]

**Review by key-joshua — APPROVED** [review]

File: server/helpers/notifications.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      description: `Hello Joshua your request was approved successfully, you will have travel on Mon 23 Feb 2020 at 16:30`, // req.request.description
+      email: 'k.joshua855@gmail.com', // req.request.email
+      requester: 'Joshua', // req.request.username
+      manager: 'agent Moriah', // req.request.agent
+    };
+  *  await notification.createNotification(clientNotify);
+
+ * required to handle
+ * viewing notification with app and send notification to email with app.
+ */
```
Comment (luc-tuyishime) [inline server/helpers/notifications.js:7]: Why these commented lines?

File: server/tests/request.test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import otherTokens from './mochData/token';
+
+chai.use(chaiHttp);
+const router = () => chai.request(app);
+
+describe('Test suite for approving/rejecting trip requests', () => {
+  it('should approve trip request if manager has user under his/her authority', async () => {
+    const res = await router().patch('/api/request/1/approve')
+      .set('token', token).send({ status: 'Approved' });
+    console.log(res.data);
```
Comment (luc-tuyishime) [inline server/tests/request.test.js]: Remove console.log every where

File: server/helpers/notifications.js (hunk 1 above)
Comment (Nigorjeanluc) [reply server/helpers/notifications.js:7]: I'm not the one who created this file

Comment (Nigorjeanluc) [reply server/helpers/notifications.js]: But I can refactor this if needed

**Pull request merged**